impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultSourceRoot.java

```diff
@@ -144,13 +143,15 @@ public DefaultSourceRoot(final ProjectScope scope, final Language language, fina
      * @param scope scope of source code (main or test)
      * @param language language of the source code
      * @param directory directory of the source code
+     * @param includes list of patterns for the files to include, or {@code null} if unspecified
```

Comment:
unspecified or nothing to include?

Reply:
Unspecified. The default is plugin-dependent. For example, the compiler plugin will default to the `*.java` includes. Other plugins may have different default.

Reply:
Actually this "null or empty" (will update the documentation). This is a parameter rather than a return value, accepting both for convenience.